pkg/webhook/mutation/pod/mutator/otlp/exporter/injectors_test.go

```diff
@@ -191,3 +196,139 @@ func TestLogsInjectorIsEnabledAndInject(t *testing.T) {
 		})
 	}
 }
+
+func TestNoProxyInjector_Inject(t *testing.T) {
+	type args struct {
+		containsNoProxy     bool
+		activeGateEnabled   bool
+		featureFlagDisabled bool
+		noProxyValue        string
+		alreadyContainsFQDN bool
+	}
+
+	const agFQDN = "dynakube-activegate.dynatrace"
+
+	makeDynakube := func(activeGateEnabled, featureFlagDisabled, hasProxy bool) *dynakube.DynaKube {
+		dk := &dynakube.DynaKube{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "dynakube",
+				Namespace: "dynatrace",
+			},
+		}
+
+		if activeGateEnabled {
+			dk.Spec.ActiveGate = activegate.Spec{
+				Capabilities: []activegate.CapabilityDisplayName{
+					activegate.MetricsIngestCapability.DisplayName,
+				},
+			}
+		}
+
+		if featureFlagDisabled {
+			dk.Annotations = map[string]string{
+				exp.OTLPInjectionSetNoProxy: "false",
+			}
+		}
+
+		if hasProxy {
+			dk.Spec.Proxy = &value.Source{Value: "proxy"}
+		}
+
+		return dk
+	}
+
+	tests := []struct {
+		name          string
+		args          args
+		expectMutated bool
+		expectValue   string
+	}{
+		{
+			name: "NO_PROXY not present",
+			args: args{
+				containsNoProxy:     false,
+				activeGateEnabled:   true,
+				noProxyValue:        "",
+				alreadyContainsFQDN: false,
+			},
+			expectMutated: true,
+			expectValue:   agFQDN,
+		},
+		{
+			name: "NO_PROXY present, empty",
+			args: args{
+				containsNoProxy:     true,
+				activeGateEnabled:   true,
+				noProxyValue:        "",
+				alreadyContainsFQDN: false,
+			},
+			expectMutated: true,
+			expectValue:   agFQDN,
+		},
+		{
+			name: "NO_PROXY present, value not containing FQDN",
+			args: args{
+				containsNoProxy:     true,
+				activeGateEnabled:   true,
+				noProxyValue:        "foo,bar",
+				alreadyContainsFQDN: false,
+			},
+			expectMutated: true,
+			expectValue:   "foo,bar," + agFQDN,
+		},
+		{
+			name: "NO_PROXY present, value already contains FQDN",
```

Comment:
One more test like this but with a preset value of `"foo," + agFQDN + ",bar" ` might be useful.

Reply:
added with [ec5d85f](https://github.com/Dynatrace/dynatrace-operator/pull/5642/commits/ec5d85fcbd810782ff3d5e533bfbdd774a8c108f)